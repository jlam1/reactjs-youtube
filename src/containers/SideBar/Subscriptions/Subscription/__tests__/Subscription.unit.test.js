import React from 'react';
import {shallow} from 'enzyme';
import {Subscription} from '../Subscription';

describe('Subscription', () => {

	test('Renders empty Subscription', () => {
		const wrapper = shallow(
		  <Subscription />
		);
		expect(wrapper).toMatchSnapshot();
	  });
	
	  test('Renders broadcasting Subscription', () => {
		const wrapper = shallow(
		  <Subscription broadcasting label='Subscription Label'/>
		);
		expect(wrapper).toMatchSnapshot();
	  });
	
	  test('Render non-broadcasting Subscription with new videos', () => {
		const wrapper = shallow(
		  <Subscription label='Subscription Label' amountNewVideos={2}/>
		);
		expect(wrapper).toMatchSnapshot();
	  });

});