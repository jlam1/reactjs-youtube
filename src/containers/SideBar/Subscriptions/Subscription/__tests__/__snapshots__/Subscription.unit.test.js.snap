// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Render non-broadcasting Subscription with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Subscription Label
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      2
    </span>
  </div>
</MenuItem>
`;

exports[`Subscription Renders broadcasting Subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Subscription Label
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription Renders empty Subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;
